core/src/lib/code-generation/code-builder/code-builder-imports.ts

```diff
@@ -0,0 +1,37 @@
+import { IQueueStep } from './queued-code-builder/queue-step';
+import { CodeBuilder } from './code-builder';
+
+type ILibraryImport = {
+    from: string;
+    elements: {
+        element: string;
+        importAs?: string;
+    }[];
+};
+
+export class CodeBuilderImports implements IQueueStep {
+
+    private imports: ILibraryImport[] = [];
+
+    add(element: string, from: string, importAs?: string): void {
+        let importLib = this.imports.find(x => x.from === from);
+        if (!importLib) {
+            importLib = { from, elements: [] };
+            this.imports.push(importLib);
+        }
+        if (importLib.elements.findIndex(x => x.element === element) === -1) {
+            importLib.elements.push({ element, importAs });
```

Comment:
importAs ist optional, würde dort also einen default value bei den Parametern vergeben wenn du es hier verwendest

Reply:
no there is no value necessary. if it is undefined, it will not be used.

Reply:
Ok, accepted.

Then the only thing is: I would put 
this.imports.push(importLib);
at the end of the function.

Somehow it makes it much more readable for me.

Right now the order is: change importLib -> push it to imports -> change importLib again and have in mind that you're also changing the object in imports that you pushed there before.

I would prefer: change importLib -> push it to imports
Much more straightforward.

If i missed that this changes the logic somehow feel free to tell me.